broker/lib/fabrik/UnlockResourcePoller.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const pubsub = require('pubsub-js');
+const eventmesh = require('../../../eventmesh');
+const CONST = require('../constants');
+const logger = require('../logger');
+const lockManager = require('./../../../eventmesh').lockManager;
+const errors = require('../errors');
+const config = require('../config');
+const NotFound = errors.NotFound;
+
+class UnlockResourcePoller {
+  static start() {
+    function poller(object, interval) {
+      const lockDetails = JSON.parse(object.spec.options);
+      return eventmesh.apiServerClient.getResource(lockDetails.lockedResourceDetails.resourceGroup, lockDetails.lockedResourceDetails.resourceType, lockDetails.lockedResourceDetails.resourceId)
+        .then((resource) => {
+          const resourceState = resource.body.status.state;
+          logger.debug(`[Unlock Poller] Got resource ${lockDetails.lockedResourceDetails.resourceId} state as `, resourceState);
+          if (resourceState === CONST.APISERVER.RESOURCE_STATE.SUCCEEDED || resourceState === CONST.APISERVER.RESOURCE_STATE.FAILED || resourceState === CONST.APISERVER.RESOURCE_STATE.ERROR) {
```

Comment:
reuse util method is operationCompleted.

Reply:
In what situations will the state be ERROR? shouldnt it be just FAILED in such scenarios? Its better not to have too many states